### PR TITLE
import dxf implementation

### PR DESCRIPTION
**Pull request by gmalinve**

### CODE REVIEW

**Review by Samuelopez-ansys — APPROVED** [review]

File: pyaedt/application/Analysis3D.py [hunk 1]
```diff
@@ -1118,3 +1118,160 @@ def check_intersections(output, input_list, cad_in=None):
                     nets["Net{}".format(k)] = [cad_to_investigate.name]
                     break
         return nets
+
+    @pyaedt_function_handler()
+    def get_dxf_layers(self, file_path):
+        # type: (str) -> list
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis3D.py:1124]:
```suggestion
        # type: (str) -> list[str]
```

Comment (MaxJPRey) [reply pyaedt/application/Analysis3D.py:1124]: Thanks for adding the type hints.

File: pyaedt/application/Analysis3D.py [hunk 2]
```diff
@@ -1118,3 +1118,160 @@ def check_intersections(output, input_list, cad_in=None):
...
+        References
+        ----------
+
+        >>> oEditor.ImportDXF
+
+        """
+        for ly in layers_list:
+            if ly not in self.get_dxf_layers(file_path):
+                self.logger.error("{} does not exist in specified dxf.".format(ly))
+                return False
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis3D.py:1224]:
```suggestion
        for layer in layers_list:
            if layer not in self.get_dxf_layers(file_path):
                self.logger.error("{} does not exist in specified dxf.".format(layer))
                return False
```

**Review by MaxJPRey — APPROVED** [review]
LGTM.
TIL. I did not know we could use the dxf import for the layers in Maxwell.

File: pyaedt/application/Analysis3D.py [hunk 3]
```diff
@@ -1118,3 +1118,160 @@ def check_intersections(output, input_list, cad_in=None):
                     nets["Net{}".format(k)] = [cad_to_investigate.name]
                     break
         return nets
+
+    @pyaedt_function_handler()
+    def get_dxf_layers(self, file_path):
+        # type: (str) -> list[str]
+        """Read a dxf file and return all layer names.
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1125]:
```suggestion
        """Read a DXF file and return all layer names.
```

File: pyaedt/application/Analysis3D.py [hunk 4]
```diff
@@ -1118,3 +1118,160 @@ def check_intersections(output, input_list, cad_in=None):
...
+
+        Parameters
+        ----------
+        file_path : str
+            Full path to the dxf file.
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1130]:
```suggestion
            Full path to the DXF file.
```

File: pyaedt/application/Analysis3D.py [hunk 5]
```diff
@@ -1118,3 +1118,160 @@ def check_intersections(output, input_list, cad_in=None):
...
+
+        Returns
+        -------
+        list
+            List of layers in dxf.
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1135]:
```suggestion
            List of layers in the DXF file.
```

File: pyaedt/application/Analysis3D.py [hunk 6]
```diff
@@ -1118,3 +1118,160 @@ def check_intersections(output, input_list, cad_in=None):
...
+        defeature_geometry=False,
+        defeature_distance=0,
+        round_coordinates=False,
+        round_num_digits=4,
+        write_poly_with_width_as_filled_poly=False,
+        import_method=1,
+        sheet_bodies_2d=True,
+    ):
+        # type: (str, list, bool, bool, float, float, bool, float, bool, int, bool, int, bool) -> bool
+        """Import dxf file.
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1164]:
```suggestion
        """Import a DXF file.
```

File: pyaedt/application/Analysis3D.py [hunk 7]
```diff
@@ -1118,3 +1118,160 @@ def check_intersections(output, input_list, cad_in=None):
...
+            Default value is ``4``.
+        write_poly_with_width_as_filled_poly : bool, optional
+            Imports wide polylines as polygons.
+            Default value is ``False``.
+        import_method : int, bool
+            ``Script`` or ``Acis``.
+            Default value is ``1`` which refers to ``Acis``.
+        sheet_bodies_2d : bool, optional
+            Import as 2D sheet bodies causes imported objects to be organized in terms of 2D sheets.
+            Default value is ``True``.
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1208]:
```suggestion
        file_path : str
            Path to the DXF file.
        layers_list : list
            List of layer names to import. To get the layers in the DXF file,
            you can call the ``get_dxf_layers`` method.
        auto_detect_close : bool, optional
            Whether to check polylines to see if they are closed.
            The default is ``True``. If a polyline is closed, the modeler
            creates a polygon in the design.
        self_stitch : bool, optional
            Whether to join multiple straight line segments to form polylines.
            The default is ``True``.
        self_stitch_tolerance : float, optional
            Self stitch tolerance value. The default is ``0``.
            If not provided by the user the default tolerance value is used.
        scale : float, optional
            Scaling factor. The default is ``0.001``. The units are ``mm``.
        defeature_geometry : bool, optional
            Whether to defeature the geometry to reduce complexity.
            The default is ``False``.
        defeature_distance : float, optional
            Defeature tolerance distance. The default is ``0``.
        round_coordinates : bool, optional
            Whether to rounds all imported data to the number
            of decimal points specified by the next parameter.
            The default is ``False``.
        round_num_digits : int, optional
            Number of digits to which to round all imported data.
            The default is ``4``.
        write_poly_with_width_as_filled_poly : bool, optional
            Imports wide polylines as polygons.
            Default value is ``False``.
        import_method : int, bool
            ``Script`` or ``Acis``.
            Default value is ``1`` which refers to ``Acis``.
        sheet_bodies_2d : bool, optional
            Import as 2D sheet bodies causes imported objects to be organized in terms of 2D sheets.
            Default value is ``True``.
```

File: pyaedt/application/Analysis3D.py [hunk 8]
```diff
@@ -1118,3 +1118,160 @@ def check_intersections(output, input_list, cad_in=None):
...
+            vArg3.append("layer_type:="), vArg3.append("signal")
+            vArg2.append(vArg3)
+        vArg1.append(vArg2)
+        self.oeditor.ImportDXF(vArg1)
+        return True
+
+    @pyaedt_function_handler()
+    def _find_indices(self, list_to_check, item_to_find):
+        # type: (list, str|int) -> list
+        """Given a list returns the list of indices for all occurrences of a given element.
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1259]:
```suggestion
        """Given a list, returns the list of indices for all occurrences of a given element.
```

File: pyaedt/application/Analysis3D.py (hunk 7 above)
Comment (gmalinve) [reply pyaedt/application/Analysis3D.py:1208]: @PipKat you are right but by "default" I meant the choice. I mean.. in AEDT if the user doesn't enable the self stich tolerance he/she has to enable the default tolerance button.

**Comment by terdol** [comment]
Hi, sheet_bodies_2d argument of import_dxf function is currently ignored I think which is confusing. It is set directly in the function body.

**Pull request merged**